File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] arr, int target) {
        int n = arr.length;

        int left = 0, right = n - 1;
        int low = -1, high = -1;

        while (left <= right) {
            int mid = left + (right - left) /2;
            if (arr[mid] == target) {
                low = mid;
                right = mid - 1;
            }

            if (arr[mid] < target) {
                left = mid + 1;

            } else if (arr[mid] > target) {
                right = mid - 1;
            }
        }
    
        left = 0;
        right = n - 1;
        while (left <= right) {
            int mid = left + ((right - left)/2);
        
            if (arr[mid] == target) {
                high = mid;
                left = mid + 1;
               
            }

            if (arr[mid] < target) {
                left = mid + 1;

            } else if (arr[mid] > target) {
                right = mid - 1;
            }
        }

        return new int[] {low, high};
    }
}

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int n = nums.size();
        vector<int> ans;
        
        int left = -1, right = n;
        while (right > left + 1 ) {
            int mid = (right + left) /2;

            if (nums[mid] < target) {
                left = mid;
            } else {
                right = mid;
            }
        }
        if ( right == n || nums[right] != target ) {
            ans.push_back(-1);
        } else {
            
            ans.push_back(right);
        }

        left = -1;
        right = n;

        while (right > left + 1 ) {
            int mid = (right + left) /2;

            if (nums[mid] <= target) {
                left = mid;
            } else {
                right = mid;
            }
        }
        if ( left == -1 || nums[left] != target) {
            ans.push_back(-1);
        } else {
            ans.push_back(left);
        }
        

        return ans;
    }
};